modules/nominees-election/src/lib.rs

```diff
@@ -232,20 +158,18 @@ pub mod module {
 		pub fn bond(origin: OriginFor<T>, #[pallet::compact] amount: Balance) -> DispatchResult {
 			let who = ensure_signed(origin)?;
 
-			let mut ledger = Self::ledger(&who);
+			let ledger = Self::ledger(&who);
 			let free_balance = T::Currency::free_balance(&who);
-			if let Some(extra) = free_balance.checked_sub(ledger.total) {
+			if let Some(extra) = free_balance.checked_sub(ledger.total()) {
 				let extra = extra.min(amount);
-				let old_active = ledger.active;
-				ledger.active += extra;
-				ensure!(
-					ledger.active >= T::MinBondThreshold::get(),
-					Error::<T, I>::BelowMinBondThreshold
-				);
-				ledger.total += extra;
+
+				let old_active = ledger.active();
+
+				let ledger = ledger.bond(extra).map_err(Into::<Error<T, I>>::into)?;
```

Comment:
IMO we should avoid variable shadowing which is bug prone. `new_ledge` might be better.

Reply:
It is actually better to do shadowing than `ledger1`, `ledger2`, `ledger3` which you will eventually lost track.
It basically implements mutation without mutation.